In [6]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [7]:
import pandas as pd

In [43]:
df = pd.read_csv('BCGX-Forage Correction.csv')

## Revenue Growth (%) and Net Income Growth (%)
Checking the revenue and net income growth over the last three fiscal years by grouping the respective companies together and seeing the change in percentages.

In [44]:
df['Revenue Growth (%)'] = df.groupby(['Company'])['Total Revenue'].pct_change() * 100
df['Net Income Growth (%)'] = df.groupby(['Company'])['Net Income'].pct_change() * 100

Displaying the current table with the findings

In [45]:
df

Company  Total Revenue    Net Income  Total Assets  Total Liabilities  \
0  Microsoft   281724000000  101832000000  411976000000       205753000000   
1  Microsoft   245122000000   88136000000  512163000000       243686000000   
2  Microsoft   211915000000   72361000000  619003000000       275524000000   
3      Tesla    81462000000   12587000000   82338000000        36440000000   
4      Tesla    96773000000   14974000000  106618000000        43009000000   
5      Tesla    97690000000    7153000000  122070000000        48390000000   
6      Apple   383285000000   96995000000  352583000000       290437000000   
7      Apple   391035000000   93736000000  364980000000       308030000000   
8      Apple   416161000000  112010000000  359241000000       285508000000   

   Cash Flow from Operating Activities  Revenue Growth (%)  \
0                          87582000000                 NaN   
1                         118548000000          -12.992148   
2                         136162000000          -13.547132   
3                          14724000000                 NaN   
4                          13256000000           18.795267   
5                          14923000000            0.947578   
6                         110543000000                 NaN   
7                         118254000000            2.021994   
8                         111482000000            6.425512   

   Net Income Growth (%)  
0                    NaN  
1             -13.449603  
2             -17.898475  
3                    NaN  
4              18.964010  
5             -52.230533  
6                    NaN  
7              -3.359967  
8              19.495178

## Asset Growth (%) and Liabilities Growth Growth (%)
Checking the asset and liabilities growth over the last three fiscal years by grouping the respective companies together and seeing the change in percentages.

In [46]:
df['Asset Growth (%)'] = df.groupby(['Company'])['Total Assets'].pct_change() * 100
df['Liabilities Growth (%)'] = df.groupby(['Company'])['Total Liabilities'].pct_change() * 100

Displaying the current table with the findings

In [47]:
df

Company  Total Revenue    Net Income  Total Assets  Total Liabilities  \
0  Microsoft   281724000000  101832000000  411976000000       205753000000   
1  Microsoft   245122000000   88136000000  512163000000       243686000000   
2  Microsoft   211915000000   72361000000  619003000000       275524000000   
3      Tesla    81462000000   12587000000   82338000000        36440000000   
4      Tesla    96773000000   14974000000  106618000000        43009000000   
5      Tesla    97690000000    7153000000  122070000000        48390000000   
6      Apple   383285000000   96995000000  352583000000       290437000000   
7      Apple   391035000000   93736000000  364980000000       308030000000   
8      Apple   416161000000  112010000000  359241000000       285508000000   

   Cash Flow from Operating Activities  Revenue Growth (%)  \
0                          87582000000                 NaN   
1                         118548000000          -12.992148   
2                         136162000000          -13.547132   
3                          14724000000                 NaN   
4                          13256000000           18.795267   
5                          14923000000            0.947578   
6                         110543000000                 NaN   
7                         118254000000            2.021994   
8                         111482000000            6.425512   

   Net Income Growth (%)  Asset Growth (%)  Liabilities Growth (%)  
0                    NaN               NaN                     NaN  
1             -13.449603         24.318650               18.436183  
2             -17.898475         20.860546               13.065174  
3                    NaN               NaN                     NaN  
4              18.964010         29.488207               18.026894  
5             -52.230533         14.492862               12.511335  
6                    NaN               NaN                     NaN  
7              -3.359967          3.516052                6.057424  
8              19.495178         -1.572415               -7.311625

# Findings and Conclusions
## Revenue Growth
Looking at the revenue growth over the 3 fiscal years, Microsoft revenue growth has been decreasing by 13% for the past 2 years. Tesla had a big increase in 2023, but only had a small increase in  2024. Apple in 2024 had a small revenue gain, but had a 4% increase in 2025.

## Net Income Growth
Microsoft net income has been decreasing over the last two fiscal years. Tesla experience a decrease in net income by over 50% in 2024 after a 18% increase in 2023. Apple experienced a small decreasae of 3% in 2024, but had an increase in 19% in the following year.

## Asset Growth
Both  Microsoft and Tesla experienced an increase in the last 2 fiscal years in their assets. Tesla had a notable drop from 29% to 14% in 2024. Apple had a small 3.5% increase in 2024 and went under by 1.5% in the following year. For all 3 companies, the percentage of their asset in the most recent fiscal year didn't match the  last year.

## Liabilities  Growth
All companies experienced a decrease  in their liabilities in the latest fiscal year. Notably, Apple had a 7% decrease in liabilities. Both Microsoft and Tesla had around a 18% increase in liabilities in their last fiscal year and  around a 13% increase in the most recent.

# Trends
Notably, the data shows that the most recent fiscal year is highlighting a decrease in overall growth for Tesla. While their revenue growth hasn't went under, their net income was decreased by over 50%. For Microsoft, their revenue growth has been steadily decreasing along with their Net Income over the last two fiscal years.